### ISSUE TITLE
Bio-Linux compatibility

### ISSUE THREAD

**Issue by tony-travis** [evoldoers/biomake#47]
I tried BioMake under Bio-Linux 8.0.8 based on Ubuntu 14.04.5, LTS but encountered problems:

# download and install
root@wildcat:/usr/local/src/bioinformatics/BioMake# 
git clone https://github.com/evoldoers/biomake.git
apt install swi-prolog
prolog --version
SWI-Prolog version 6.6.4 for amd64

# test
root@wildcat:/usr/local/src/bioinformatics/BioMake/biomake# 
make |& tee make.log
git clean -fd t/target || true
/usr/local/src/bioinformatics/BioMake/biomake/bin/swipl_wrap -q -t test -l prolog/test/test
ERROR: /usr/lib/swi-prolog/library/prolog_stack.pl:458:
	dynamic/1: No permission to modify static procedure `prolog_exception_hook/4'
	Defined at /usr/local/src/bioinformatics/BioMake/biomake/prolog/test/test.pl:23
   [8] backtrace(99) at /usr/lib/swi-prolog/library/prolog_stack.pl:376
   [7] prolog_exception_hook(error(...,...),_G810,97,none) at /usr/local/src/bioinformatics/BioMake/biomake/prolog/test/test.pl:25
   [6] <meta call>
   [5] string_chars([70|...],_G867) <foreign>
   [4] announce([70|...]) at /usr/local/src/bioinformatics/BioMake/biomake/prolog/test/test.pl:319
   [3] test at /usr/local/src/bioinformatics/BioMake/biomake/prolog/test/test.pl:36
ERROR: announce/1: Undefined procedure: string_chars/2
   Exception: (5) string_chars([70, 65, 73, 76, 85, 82, 69, 32|...], _G775) ? EOF: exit

# check versions of SWI-Prolog available in Ubuntu repo's
root@wildcat:/usr/local/src/bioinformatics/BioMake/biomake# apt-cache policy swi-prolog
swi-prolog:
  Installed: 6.6.4-2ubuntu1
  Candidate: 6.6.4-2ubuntu1
  Version table:
 *** 6.6.4-2ubuntu1 0
        500 http://gb.archive.ubuntu.com/ubuntu/ trusty/universe amd64 Packages
        100 /var/lib/dpkg/status
     5.10.4-3ubuntu1 0
        500 http://archive.ubuntu.com/ubuntu/ precise/universe amd64 Packages

Any ideas?

  Tony.

**Comment by cmungall** [comment]
I don't believe we've tested under swipl6, swipl7 has been out for a while now.

Does this help?
http://www.swi-prolog.org/build/PPA.html
